platforms/android/tangram/src/main/java/com/mapzen/tangram/TouchInput.java

```diff
@@ -335,6 +361,11 @@ public boolean onTouch(View v, MotionEvent event) {
         shoveGestureDetector.onTouchEvent(event);
         rotateGestureDetector.onTouchEvent(event);
 
+        // Infer ending of a onScroll/onFling (pan) gesture
+        if (detectedGestures.contains(Gestures.PAN) && event.getActionMasked() == MotionEvent.ACTION_UP) {
+            onRegionDidChangeAnimated(true);
```

Comment:
Performing logic about the map in `TouchInput` ignores the behavior of any client touch responders, which could "consume" any events and prevent the map from changing. That is to say: we don't actually know whether a gesture will cause the map to change until after we've run the client responder.

Reply:
By the way I'm open to changing this relationship with client touch responders - I think it's kind of cumbersome to work with as is.